zerver/webhooks/stripe/view.py

```diff
@@ -198,6 +186,70 @@ def default_body(update_blacklist: List[str]=[]) -> str:
             amount=amount,
             end=end
         )
+
+    if category.startswith('invoice'):
+        object_id = object_['id']
+        link = "https://dashboard.stripe.com/invoices/{}".format(object_id)
+        amount = amount_string(object_["amount_due"], object_["currency"])
+        body_t = "An {invoice_type} **[invoice]({link})** for the payment of amount **{amount}** {end}."
+
+        if event == "created":
+            end = "has been created"
+            invoice_type = ""
+            topic = "invoice {}".format(object_id)
+
+        if event == "deleted":
+            end = "has been deleted"
+            invoice_type = ""
+            topic = "invoice {}".format(object_id)
+
+        if event == "finalized":
+            end = "has been finalized"
+            invoice_type = ""
+            topic = "invoice {}".format(object_id)
+
+        if event == "marked_uncollectible":
+            end = "has been marked as uncollectible"
+            invoice_type = ""
+            topic = "invoice {}".format(object_id)
+
+        if event == "payment_failed":
+            end = "has failed"
+            invoice_type = ""
+            topic = "invoice {}".format(object_id)
+
+        if event == "payment_succeeded":
+            end = "has succeeded"
+            invoice_type = ""
+            topic = "invoice {}".format(object_id)
+
+        if event == "sent":
+            end = "has been sent"
+            invoice_type = ""
+            topic = "invoice {}".format(object_id)
+
+        if event == "upcoming":
+            end = "has been created"
+            invoice_type = "upcoming"
+            topic = "invoice {}".format(object_id)
+
+        if event == "updated":
+            end = "has been updated"
+            invoice_type = ""
+            topic = "invoice {}".format(object_id)
+
+        if event == "voided":
+            end = "has been voided"
+            invoice_type = ""
+            topic = "invoice {}".format(object_id)
+
+        body = body_t.format(
+            invoice_type = invoice_type,
+            link=link,
+            amount=amount,
+            end=end
+        )
+
```

Comment:
Okay, so there is a better way to write this. This is a lot of repetitive code. A better way to write it is to have a dictionary at the top of the file that matches an `event` to its `end` text. Let's call the dictionary `INVOICE_INFO`. After we do that the code becomes just:

```
end = INVOICE_INFO.get(event)
```

And you can change `invoice_type` manually, that's fine!

Reply:
Done!